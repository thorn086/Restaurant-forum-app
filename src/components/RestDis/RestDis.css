.add{
width: auto;}

.none{
    color: white;
    text-decoration: uppercase;
}
@media (min-width: 760px){
    .back-btn.rest,.back-btn.add{
        margin:2rem 0 2rem 6.3rem
    }
}
@media (min-width: 1020px){
    .state_list.rest{
        width: 80%;
    }
    .back-btn.rest,.back-btn.add{
        margin:4rem 0 4rem 7rem
    }
}
@media (min-width: 1020px){
    .state_list.rest{
        width: 70%;
    }
    .back-btn.rest,.back-btn.add{
        margin:4rem 0 4rem 11rem
    }
}
@media (min-width: 1950px){
    .state_list.rest{
        width: 60%;
    }
    .back-btn.rest,.back-btn.add{
        margin:4rem 0 4rem 15rem
    }
}
@media (min-width: 1950px){
    .state_list.rest{
        width: 40%;
    }
    .back-btn.rest,.back-btn.add{
        margin:4rem 0 4rem 12.5rem
    }
}