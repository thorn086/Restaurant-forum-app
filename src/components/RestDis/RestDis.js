import React from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import RestForumContext from '../../context'
import config from '../../config'
import AddRestaurant from '../AddRestaurant/AddRestaurant'
import './RestDis.css'

class RestDis extends React.Component {
    static defaultProps = {
        match: {
            params: {}
        }
    }
    static contextType = RestForumContext
    state={
        show:false
    }

    componentDidMount() {
        const { id } = this.props.match.params
        fetch(`${config.API_ENDPOINT}/city/${id}`)
            .then(res => {
                if (res.ok)
                    return res.json()
                throw new Error(res.statusText)

            })
            .then(restaurants => this.context.addRestaurants(restaurants))
            .catch(error => {
                this.setState({
                    error: error.message
                })
            })
    }
    showRestaurantAdd(){
        this.setState({
            show:!this.state.show
        })
    }


    handleRestaurantTitle() {
        const {show} =this.state
        if(show === false){
            return (
                'Add Restaurant'
            )
        }else{
            return(
                'Cancel'
            )
        }

    }

    handleHasRestaurants(){
        const {restaurants}  = this.context
        if(restaurants.length === 0){
            return(
                <ul className='state-list-items'>
                     <li className='state-items none'>
                        There are No Restaurants in this City, You can Add One!
                     </li>
                </ul>
            )
        }
        return(
            <ul className='state-list-items'>
            {restaurants.map((newRest,i)=>{
                return(
                <li key={i} className='state-items'>
                    <NavLink className='state-list-link'
                        to={`/restaurant/${newRest.id}`}>
                        <h3 className='state-name'>
                            {newRest.name}
                        </h3>
                    </NavLink>
                </li>
                )}
            )}
        </ul>
        )
    }

    render() {
       


        return (
            <div className='state_list rest'>
                {this.handleHasRestaurants()}
                {this.state.show ? <AddRestaurant /> :null}
                <button className=' back-btn rest' tag='button' onClick={() => this.props.history.goBack()}>Back</button>
                <button className=' back-btn add' tag='button' onClick={()=>this.showRestaurantAdd()}>{this.handleRestaurantTitle()}</button>

            </div>
        )
    }
}
export default withRouter(RestDis)