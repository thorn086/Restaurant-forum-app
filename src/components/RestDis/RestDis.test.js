import React from 'react';
import RestDis from './RestDis';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('RestDis', () => {
    const restDis = shallow(<RestDis/>);
  
    it('renders properly', () => {
        expect(restDis).toMatchSnapshot();
    });
});