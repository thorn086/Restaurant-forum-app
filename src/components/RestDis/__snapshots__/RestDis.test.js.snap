// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestDis renders properly 1`] = `ShallowWrapper {}`;
